### DISCUSSION TITLE
How to monitor WebView restarts in production?

### DISCUSSION THREAD

**Discussion by MattiasMartens** [ionic-team/capacitor#7216] [Q&A]
Hi there,

I am working with a team that is debugging a production situation where the app becomes unusable on iOS for certain users. Based on what we are seeing, it appears the cause is constant restarting of the web view, probably prompted by an issue with WebKit similar to [this one](https://github.com/ionic-team/ionic-framework/issues/25760).

We’re working on the fix, but we also want to get visibility into when and how often such crashes are occurring, as well as basic debugging information like where the user was in the app when it happened. The best I have been able to find so far for this purpose is:

```
App.addListener('appStateChange', ...)
```

...but of course this only gives me the boolean `isActive`. Which tells us if the app restarted but doesn’t tell us why. It might have been normal user action.

So my question here is twofold:
1. Using Capacitor, is there a way to distinguish between a browser environment that is starting up normally, or due to restarting after a memory highwater event or crash?
2. Are there perhaps solutions to this at the native code level? e.g. could we bundle code with the iOS app that detects when the UIWebview process has chosen to reboot WKWebview, and report it to our analytics? Or is UIWebview process a black box from that perspective?

Any help or added context would be greatly appreciated.

**Comment by bluewilliams** [comment]
Would love to hear more about where you ended up landing with this.

**Comment by mrobst** [comment]
Did you get anywhere with this? I'm having similar issues but not real visibility on iOS
